platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -448,6 +448,37 @@ MGL_EXPORT
 
 - (void)putResourceWithUrl:(NSURL *)url data:(NSData *)data modified:(nullable NSDate *)modified expires:(nullable NSDate *)expires etag:(nullable NSString *)etag mustRevalidate:(BOOL)mustRevalidate __attribute__((deprecated("", "-preloadData:forURL:modificationDate:expirationDate:eTag:mustRevalidate:")));
 
+
+/**
+A block to be called once the data  has been preloaded.
+*/
+typedef void (^MGLOfflinePreloadDataCompletionHandler)(void);
+
+/**
+Inserts the provided resource into the ambient cache.
```

Comment:
Distinguish this method from the original one:

```suggestion
Inserts the provided resource into the ambient cache, calling a completion handler when finished.
```

And add a cross-reference like this to the original method’s documentation comment, to the paragraph about asynchronicity:

> To find out when the resource is ready to retrieve from the cache, use the `-preloadData:forURL:modificationDate:expirationDate:eTag:mustRevalidate:completionHandler:` method.

Reply:
done